tardis/tardis_portal/models/datafile.py

```diff
@@ -161,8 +161,12 @@ def save(self, *args, **kwargs):
                 not self.md5sum and not self.sha512sum and require_checksums:
             raise Exception('Every Datafile requires a checksum')
         elif settings.REQUIRE_DATAFILE_SIZES and \
-                not self.size:
+             str(self.size).strip() is '':
```

Comment:
If `self.size is None`, then `str(self.size) == 'None'`. How about:
```
elif settings.REQUIRE_DATAFILE_SIZES and (
   (isinstance(self.size, str) and (self.size.strip() == '' or long(self.size.strip()) < 0)) or 
   (isinstance(self.size, (int, long)) and self.size < 0)):
```
then change the Exception text to include the negative size and get rid of the other test?